### PR TITLE
Feature/search filter

### PR DESCRIPTION
**Pull request by JamesSCBD**
search filter

### CODE REVIEW

File: components/List/Columns/Email.vue [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    {{ getEmailAddress(email) }}
+  </div>
+</template>
+
+<script>
+// import CountryFlag from 'vue-country-flag'
+
+export default {
+  components: {  },
+  props     : { email: { type: Array, required: true } },
```
Comment (sbilodeau) [inline components/List/Columns/Email.vue]: If more than one than make it plural: `emails`

Comment (JamesSCBD) [reply components/List/Columns/Email.vue]: done

File: pages/_conferenceCode/contacts/index.vue [hunk 2]
```diff
@@ -11,32 +274,145 @@
...
 // ===================
 function initData (){
   return {
-    loading: false
+    loading          : false,
+    sizes            : [ 'Small', 'Medium', 'Large', 'Extra Large' ],
+    countryOptions   : [],
+    flagOptions      : [ { name: 'Funded', value: 1 }, { name: 'priority', value: 2 } ],
+    attendanceOptions: [
+      { name: 'Any', value: 0, code: 'Any' },
```
Comment (sbilodeau) [inline pages/_conferenceCode/contacts/index.vue]: `Any` should nerver be an option you should have a `x Clear` as first option

File: pages/_conferenceCode/contacts/index.vue [hunk 3]
```diff
@@ -11,32 +274,145 @@
...
+      { name: 'Nominated', value: 1 << 0, code: 'Nom' },
+      { name: 'Accredited', value: 1 << 1, code: 'Acc' },
+      { name: 'Registered', value: 1 << 2, code: 'Reg' }
+    ],
+    scopeOptions: [
+      { name: 'Any', value: 1 },
```
Comment (sbilodeau) [inline pages/_conferenceCode/contacts/index.vue]: `Any` should nerver be an option you should have a `x Clear` as first option
NOT mutiple selection field

File: pages/_conferenceCode/contacts/index.vue [hunk 4]
```diff
@@ -11,32 +274,145 @@
...
+      { name: 'Goverment', value: 2 },
+      { name: 'Country/Address', value: 3 }
```
Comment (sbilodeau) [inline pages/_conferenceCode/contacts/index.vue]:
```suggestion
      { name: 'Country (Address)', value: 3 }
```

File: pages/_conferenceCode/contacts/index.vue [hunk 5]
```diff
@@ -1,8 +1,271 @@
...
+      <div class="row">
+        <div class="col-md-6 col-sm-6 col-xs-12">
+          <div class="row">
+            <div class="col-md-8 col-sm-7 col-xs-12">
+              <div class="form-group">
+                <BFormInput
+                  id="filterInput"
+                  v-model="filter.name"
+                  type="search"
+                  placeholder="Name /Barcode /#tags"
```
Comment (sbilodeau) [inline pages/_conferenceCode/contacts/index.vue]:
```suggestion
                  placeholder="Name / Barcode / #tags"
```

File: pages/_conferenceCode/contacts/index.vue [hunk 6]
```diff
@@ -1,8 +1,271 @@
...
+      </div>
+      <div class="row">
+        <div class="col-md-6 col-sm-6 col-xs-12">
+          <div class="form-group">
+            <multiselect
+              id="ajax"
+              v-model="filter.selectedOrganization"
+              label="OrganizationName"
+              track-by="OrganizationUID"
+              placeholder="Organization(autocomplete)"
```
Comment (sbilodeau) [inline pages/_conferenceCode/contacts/index.vue]:
```suggestion
              placeholder="Organization"
```

File: pages/_conferenceCode/contacts/index.vue [hunk 7]
```diff
@@ -1,8 +1,271 @@
...
+        <div class="col-md-6 col-sm-6 col-xs-12">
+          <div class="row">
+            <div class="col-md-7 col-sm-7 col-xs-12 pr-0">
+              <div class="form-group">
+                <multiselect
+                  id="ajax"
+                  v-model="filter.selectedMeetings"
+                  label="Code"
+                  track-by="Code"
+                  placeholder="Meetings attendance"
```
Comment (sbilodeau) [inline pages/_conferenceCode/contacts/index.vue]:
```suggestion
                  placeholder="Meetings"
```

File: pages/_conferenceCode/contacts/index.vue [hunk 8]
```diff
@@ -11,32 +274,145 @@
...
+  },
+  data    : initData,
+  computed: { totalRows: rows },
+  mounted,
+  methods : {
+    tableItems: search,
+    clearAllMeetings,
+    getMeetingsList,
+    getAttendanceValue,
+    clearAllAttendance,
```
Comment (sbilodeau) [inline pages/_conferenceCode/contacts/index.vue]: Why the Mutiliselect field is not managing the clear function?

File: pages/_conferenceCode/contacts/index.vue [hunk 9]
```diff
@@ -11,32 +274,143 @@
...
+    attendanceOptions: [
+      { name: 'Any', value: 0, code: 'Any' },
+      { name: 'Nominated', value: 1 << 0, code: 'Nom' },
+      { name: 'Accredited', value: 1 << 1, code: 'Acc' },
+      { name: 'Registered', value: 1 << 2, code: 'Reg' }
+    ],
+    scopeOptions: [
+      { name: 'Any', value: 1 },
+      { name: 'Goverment', value: 2 },
+      { name: 'Country/Address', value: 3 }
```
Comment (sbilodeau) [inline pages/_conferenceCode/contacts/index.vue]:
```suggestion
      { name: 'Goverment', value: 'GOV' },
      { name: 'Country (Address)', value: 'CTR' }
```

File: pages/_conferenceCode/contacts/index.vue (hunk 8 above)
Comment (JamesSCBD) [reply pages/_conferenceCode/contacts/index.vue]: Create a single method for all multiselect

File: pages/_conferenceCode/contacts/index.vue (hunk 3 above)
Comment (JamesSCBD) [reply pages/_conferenceCode/contacts/index.vue]: We show x clear option near up/down caret for all single and multiple selection filelds

File: pages/_conferenceCode/contacts/index.vue (hunk 2 above)
Comment (JamesSCBD) [reply pages/_conferenceCode/contacts/index.vue]: We show x clear option near up/down caret for all single and multiple selection filelds

File: pages/_conferenceCode/contacts/index.vue [hunk 10]
```diff
@@ -47,29 +299,84 @@ async function search (ctx){
...
+    skip                   : skipRecord,
+    FreeText               : filter.name,
+    EventRegistrationStatus: getAttendanceValue(filter),
+    StatusForEventUID1     : getMeetingsIds(filter, 0),
+    StatusForEventUID2     : getMeetingsIds(filter, 1),
+    StatusForEventUID3     : getMeetingsIds(filter, 2),
+    StatusForEventUID4     : getMeetingsIds(filter, 3),
+    OrganizationUIDs       : getOrganizationIds(filter),
+    Governments            :
+      filter.selectedCountry === '' || filter.selectedCountry === null ? '' : [ filter.selectedCountry.Code ],
```
[the remark this answers is not in the document]
Comment (JamesSCBD) [reply pages/_conferenceCode/contacts/index.vue]: I have a already done

[the remark this answers is not in the document]
Comment (JamesSCBD) [reply pages/_conferenceCode/contacts/index.vue]: Done

File: pages/_conferenceCode/contacts/index.vue (hunk 8 above)
Comment (sbilodeau) [reply pages/_conferenceCode/contacts/index.vue]: I was thinking about a multiselect component that you passe le selectable element, the compnent binding returns the selected objects and the clear() function is generic

**Pull request merged**